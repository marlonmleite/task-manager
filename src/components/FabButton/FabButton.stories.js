import React from 'react'
import { storiesOf } from '@storybook/react'
import FabButton from './'

const storie = storiesOf('FabButton', module)

storie.add('default', () => <FabButton color="default" />)
storie.add('primary', () => <FabButton color="primary" />)
storie.add('secondary', () => <FabButton color="secondary" />)
storie.add('info', () => <FabButton color="info" />)
storie.add('danger', () => <FabButton color="danger" />)
