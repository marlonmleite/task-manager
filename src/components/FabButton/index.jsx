import React, { Component } from 'react'
import PlusIcon from 'core/assets/svg/plus.svg'
import FabButtonStyled from './styled'

class FabButton extends Component {

  static defaultProps = {
    color: 'primary',
  }

  render() {
    return (
      <FabButtonStyled {...this.props}>
        <PlusIcon fill="#fff" />
      </FabButtonStyled>
    )
  }

}

export default FabButton
