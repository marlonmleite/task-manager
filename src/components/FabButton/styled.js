import styled from 'styled-components'
import Button from 'components/Button'

const FabButton = styled(Button)`
  padding: 1rem;
  border-radius: 100%;
  width: 56px;
  height: 56px;
  position: fixed;
  right: 20px;
  bottom: 25px;
  box-shadow: 0 0 4px rgba(0,0,0,.04), 0 4px 8px rgba(0,0,0,.08);
`

export default FabButton
