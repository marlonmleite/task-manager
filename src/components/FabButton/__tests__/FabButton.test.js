import React from 'react'
import { shallow } from 'enzyme'
import FabButton from '../'

test('should render FabButton', () => {
  const fab = shallow(<FabButton>fab</FabButton>)

  expect(fab).toMatchSnapshot()
})
