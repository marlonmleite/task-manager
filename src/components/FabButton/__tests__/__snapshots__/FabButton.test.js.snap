// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FabButton 1`] = `
<Styled(Button)
  color="primary"
>
  <[object Object]
    fill="#fff"
  />
</Styled(Button)>
`;
